modules/deployments-comparing-deploymentconfigs.adoc

```diff
@@ -0,0 +1,112 @@
+// Module included in the following assemblies:
+//
+// * applications_and_projects/deployments/what-deployments-are.adoc
+
+[id='deployments-comparing-deploymentconfigs-{context}']
+= Comparing Deployments and DeploymentConfigs
+
+Because deployment configurations existed in {product-title} prior to
+deployments being added in Kubernetes 1.2, the latter object type naturally
+diverges slightly from the former. The long-term goal in {product-title} is to reach
+full feature parity in Deployments and switch to using them as a
+single object type that provides fine-grained management over applications.
+
+Deployments are supported to ensure upstream projects and examples that use the
+new object type can run smoothly on {product-title}. Given the current feature
+set of Deployments, you might want to use them instead of deployment
+configurations in {product-title} if you do not plan to use any of the following
+in particular:
+
+- Image streams
+- Lifecycle hooks
+- Custom deployment strategies
+
+The following sections go into more details on the differences between the two
+object types to further help you decide when you might want to use Kubernetes
+deployments over deployment configurations.
+
+[id='delpoymentconfigs-specific-features-{context}']
+== DeploymentConfigs-specific features
+
+[discrete]
+==== Automatic rollbacks
+
+Currently, Deployments do not support automatically rolling back to the last
+successfully deployed ReplicaSet in case of a failure.
+
+[discrete]
+==== Triggers
+
+Deployments have an implicit `ConfigChange` trigger in that every
+change in the pod template of a deployment automatically triggers a new rollout.
+If you do not want new rollouts on pod template changes, pause the deployment:
+
+----
+$ oc rollout pause deployments/<name>
+----
+
+Currently, Deployments do not support `ImageChange` triggers. A generic
+triggering mechanism has been proposed upstream, but it is unknown if and when
+it may be accepted. Eventually, a {product-title}-specific mechanism could be
+implemented to layer on top of Deployments, but it would be more desirable for
+it to exist as part of the Kubernetes core.
+
+[discrete]
+==== Lifecycle hooks
+
+Deployments do not support any lifecycle hooks.
+
+[discrete]
+==== Custom strategies
+
+Deployments do not yet support user-specified Custom deployment
+strategies yet.
+
+[discrete]
+==== Canary deployments
+
+Deployments do not yet run canaries as part of a new rollout.
```

Comment:
@tnozicka So do you think that the whole "canary deployments" section should be removed here? We do already have an "automatic rollbacks" section that covers what you mention.

Reply:
I guess it's a bit tricky and I am not sure what the author's intention was. But since the doc isn't out yet, i think the safe choice is to remove it and we can add it back rephased if we see fit?

Reply:
Sounds good to me.